src/xSTIR/pSTIR/STIR.py

```diff
@@ -2336,6 +2336,135 @@ def __del__(self):
         if self.handle is not None:
             pyiutil.deleteDataHandle(self.handle)
 
+    def set_kappa(self, image):
+        """Sets kappa."""
+        assert_validity(image, ImageData)
+        parms.set_parameter(self.handle, 'QuadraticPrior', 'kappa', image.handle)
+
+    def get_kappa(self):
+        """Returns kappa."""
+        image = ImageData()
+        image.handle = pystir.cSTIR_parameter(self.handle, 'QuadraticPrior', 'kappa')
+        check_status(image.handle)
+        return image
+
+
+class LogcoshPrior(Prior):
+    r"""Class for Log-cosh Prior.
+
+    Implements the prior, Log-cosh Prior, one of the earliest uses in P. J. 
+    Green's paper "Bayesian reconstructions from emission tomography data using 
+    a modified EM algorithm," in IEEE Transactions on Medical Imaging, vol. 9, 
+    no. 1, pp. 84-93, March 1990, doi: 10.1109/42.52985.
+
+    The prior has one parameter the scalar, it is the edge-preservation parameter.
+
+    The log-cosh function is given by:
+    \f[
+        f = \sum_{dr} w_{dr} 1/scalar^2 * log(cosh(\lambda_r - \lambda_{r+dr})) * \kappa_r * \kappa_{r+dr}))
```

Comment:
this is still incorrect.
```suggestion
        f_r = \sum_{r,dr} w_{dr} \frac{1}{2 s^2}  log(cosh(s(\lambda_r - \lambda_{r+dr}))) * \kappa_r * \kappa_{r+dr}$
```